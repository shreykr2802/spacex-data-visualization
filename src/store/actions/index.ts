export { fetchDataStart, fetchataSuccess, fetchDataFailed } from './spaceXActions';