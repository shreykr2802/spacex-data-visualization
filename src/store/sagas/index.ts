import { takeEvery } from "redux-saga/effects";
import spaceXDataSaga from "./spaceXSaga";
import { fetchDataStart } from "../slices/spaceXSlice";

export function* watchSpaceXData() {
  yield takeEvery(fetchDataStart, spaceXDataSaga);
}

export default watchSpaceXData;
