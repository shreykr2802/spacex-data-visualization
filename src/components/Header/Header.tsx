import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { setSearchTerm } from "../../store/slices/searchSlice";
import { RootState } from "../../store/store";
import "materialize-css/dist/css/materialize.min.css";
import SideNavButton from "../SideNav/SideNavButton";
const M = require("materialize-css/dist/js/materialize.min.js");

const filterInfo: any = {
  lastweek: 'Launched Last Week',
  lastmonth: 'Launched Last Month',
  lastyear: 'Launched Last Year',
  failure: 'Failed to Launched',
  success: 'Launched Successfully',
  yes: 'Upcoming',
  no: 'Completed'
}

const Header = () => {
  const dispatch = useDispatch();
  const searchTerm = useSelector((state: RootState) => state.search.searchTerm);
  const filterBy = useSelector((state: RootState) => state.search.filterBy);

  useEffect(() => {
    var elem = document.querySelector(".sidenav");
    // eslint-disable-next-line
    var instance = M.Sidenav.init(elem, {
      edge: "right",
      inDuration: 250
    });
  }, []);

  return (
    <div className="navbar-fixed row">
      <nav className="nav-wrapper deep-purple">
        <div className="input-field col s9">
          <input
            id="search"
            type="search"
            required
            value={searchTerm}
            onChange={(event) => dispatch(setSearchTerm(event.target.value))}
            placeholder="Search by Rocket Name"
          />
          <label className="label-icon">
            <i className="material-icons">search</i>
          </label>
          <i className="material-icons" onClick={() => dispatch(setSearchTerm(""))}>
            close
              </i>

        </div>
        <div className="col s2">
          <span className="filerInfo">{filterInfo[filterBy]}</span>
        </div>
        <div className="col s1">
          <SideNavButton />
        </div>
      </nav>
    </div>
  );
};

export default Header;
