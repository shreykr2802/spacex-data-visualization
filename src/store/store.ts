import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import createSagaMiddleware from "redux-saga";

import spaceXReducer from "./slices/spaceXSlice";
import { watchSpaceXData } from "./sagas/index";

const rootReducer = combineReducers({
  spaceX: spaceXReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(thunk, sagaMiddleware))
);

sagaMiddleware.run(watchSpaceXData);

export default store;
