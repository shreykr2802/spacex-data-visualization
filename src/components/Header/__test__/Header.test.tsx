import React from 'react';
import ReactDom from 'react-dom';
import { Provider } from 'react-redux';
import store from '../../../store/store';
import Header from '../Header';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Provider store={store}><Header /></Provider>, div);
});

it("matches snapshot", () => {
    const tree = renderer.create(<Provider store={store}><Header /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})