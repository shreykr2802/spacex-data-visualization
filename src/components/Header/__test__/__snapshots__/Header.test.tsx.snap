// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar-fixed row"
>
  <nav
    className="nav-wrapper deep-purple"
  >
    <div
      className="input-field col s9"
    >
      <input
        id="search"
        onChange={[Function]}
        placeholder="Search by Rocket Name"
        required={true}
        type="search"
        value=""
      />
      <label
        className="label-icon"
      >
        <i
          className="material-icons"
        >
          search
        </i>
      </label>
      <i
        className="material-icons"
        onClick={[Function]}
      >
        close
      </i>
    </div>
    <div
      className="col s2"
    >
      <span
        className="filerInfo"
      />
    </div>
    <div
      className="col s1"
    >
      <a
        className="top-nav sidenav-trigger show-on-large right"
        data-target="slide-out"
        href="#"
      >
        <i
          className="material-icons"
        >
          filter_list
        </i>
      </a>
    </div>
  </nav>
</div>
`;
